lib/events/dynamoevents/legacy.go

```diff
@@ -0,0 +1,148 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package dynamoevents
+
+import (
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/gravitational/trace"
+)
+
+// convertLegacyAttributesMap converts a map of legacy (SDK V1) Dynamo attribute
+// values into SDK V2 attribute values.
+func convertLegacyAttributesMap(m map[string]*LegacyAttributeValue) (map[string]types.AttributeValue, error) {
+	ret := make(map[string]types.AttributeValue)
+	for name, legacyValue := range m {
+		val, err := convertLegacyAttributeValue(legacyValue)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		ret[name] = val
+	}
+
+	return ret, nil
+}
+
+// convertLegacyAttributeValue converts a legacy (SDK V1) Dynamo attribute value
+// into a newer attribute value (SDK V2).
+func convertLegacyAttributeValue(legacyAttr *LegacyAttributeValue) (types.AttributeValue, error) {
+	switch {
+	case legacyAttr.B != nil:
+		return &types.AttributeValueMemberB{Value: legacyAttr.B}, nil
+	case legacyAttr.BOOL != nil:
+		return &types.AttributeValueMemberBOOL{Value: *legacyAttr.BOOL}, nil
+	case legacyAttr.BS != nil:
+		return &types.AttributeValueMemberBS{Value: legacyAttr.BS}, nil
+	case legacyAttr.L != nil:
+		attrs := make([]types.AttributeValue, len(legacyAttr.L))
+		for i, subAttr := range legacyAttr.L {
+			subAttrVal, err := convertLegacyAttributeValue(subAttr)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			attrs[i] = subAttrVal
+		}
+
+		return &types.AttributeValueMemberL{Value: attrs}, nil
+	case legacyAttr.M != nil:
+		attrs := make(map[string]types.AttributeValue, len(legacyAttr.M))
+		for name, subAttr := range legacyAttr.M {
+			subAttrVal, err := convertLegacyAttributeValue(subAttr)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			attrs[name] = subAttrVal
+		}
+
+		return &types.AttributeValueMemberM{Value: attrs}, nil
+	case legacyAttr.N != nil:
+		return &types.AttributeValueMemberN{Value: *legacyAttr.N}, nil
+	case legacyAttr.NS != nil:
+		return &types.AttributeValueMemberNS{Value: toStrings(legacyAttr.NS)}, nil
```

Comment:
nit:
```suggestion
		return &types.AttributeValueMemberNS{Value: aws.ToStringSlice(legacyAttr.NS)}, nil
```
same for SS

Reply:
I couldn't find this function definition. Looking at the repo, [it seems to only be mentioned in the comment](https://github.com/aws/aws-sdk-go-v2/blob/d700681bccfc0e541af2dc38715a034b37bba4e4/aws/doc.go#L43). Is this located somewhere else?

Reply:
https://github.com/aws/aws-sdk-go-v2/blob/d700681bccfc0e541af2dc38715a034b37bba4e4/aws/from_ptr.go#L59-L64 ?